app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesRepository.scala

```diff
@@ -39,18 +43,58 @@ class VulnerabilitiesRepository @Inject()(
     IndexModel(Indexes.ascending("teams"), IndexOptions().name("teams").background(true)),
   ),
 ) {
-  def search(service: Option[String] = None, id: Option[String] = None, description: Option[String] = None, team: Option[String] = None, requiresAction: Option[Boolean] = None): Future[Seq[Vulnerability]] = {
+  def search(service: Option[String] = None, id: Option[String] = None, description: Option[String] = None, team: Option[String] = None): Future[Seq[Vulnerability]] = {
     val filters = Seq(
              service.map(s => Filters.equal("service", s)),
                   id.map(i => Filters.regex("id", i)),
          description.map(d => Filters.regex("description", d)),
-                team.map(t => Filters.equal("teams", t)),
-      requiresAction.map(r => Filters.equal("requiresAction", r))
+                team.map(t => Filters.equal("teams", t))
     ).flatten
 
     filters match {
       case Nil => collection.find().toFuture()
       case more => collection.find(Filters.and(more:_*)).toFuture()
     }
   }
+
+  // use a different view to allow distinctVulnerabilitiesSummary to return a different case class
+  private val vcsCollection: MongoCollection[VulnerabilityCountSummary] =
+    CollectionFactory.collection(
+      mongoComponent.database,
+      "vulnerabilities",
+      VulnerabilityCountSummary.mongoFormat
+    )
+
+  def distinctVulnerabilitiesSummary(id: Option[String], requiresAction: Option[Boolean]): Future[Seq[VulnerabilityCountSummary]] = {
+
+    val optFilters = Seq(
+                  id.map(i => Filters.regex("id", i)),
+      requiresAction.map(r => Filters.equal("requiresAction", r))
+    ).flatten
+
+    val pipeline = Seq(
+      group(id = "$id", push("teams", "$teams"), addToSet("services", "$service"), first("distinctVulnerability", "$$ROOT")),
+      project(
+        BsonDocument(
+          "servicesCount" -> BsonDocument("$size" -> "$services"),
+          "teams" -> BsonDocument("$reduce" ->
+            BsonDocument(
+              "input" -> "$teams",
+              "initialValue" -> BsonArray(),
+              "in" -> BsonDocument("$concatArrays" -> BsonArray("$$value", "$$this"))
+            )
+          ),
+          "distinctVulnerability" -> "$distinctVulnerability"
+      )),
+      unwind("$teams", model.UnwindOptions().preserveNullAndEmptyArrays(true)),
+      group(id = "$_id", addToSet("teams", "$teams"), first("servicesCount", "$servicesCount"), first("distinctVulnerability", "$distinctVulnerability"))
+    )
+
+    val finalPipeline = optFilters match {
```

Comment:
You can keep the first optional pipeline in the position it's applied - i.e. before the rest of the steps:
e.g.
```scala
val pipeline = optFilters.fold(Seq(`match`(Filters.and(more:_*)))(Seq.empty) ++ 
```